core/model.py

```diff
@@ -46,20 +47,10 @@ def _predict(self, x):
         # this model does not like punctuation touching characters
         x = re.sub('([.,!?()])', r' \1 ', x)  # https://stackoverflow.com/a/3645946/
 
-        os.system("python lm_1b/lm_1b_eval.py --mode sample \
-                                     --prefix \"" + x + "\" \
-                                     --pbtxt assets/data/graph-2016-09-10.pbtxt \
-                                     --vocab_file assets/data/vocab-2016-09-10.txt  \
-                                     --ckpt 'assets/data/ckpt-*' \
-                                     --num_samples 1")
-
         try:
-            # if the above os.system command fails then "out.txt" is never created
-            txt_file = open("out.txt", "r")
+            output = lm_1b_eval.generate(x)
         except OSError:
             print("Error generating a prediction, this is likely due to a lack of memory allocated to Docker")
             print("The minimum recommended resources for this model is 8 GB Memory and 4 CPUs")
         else:
-            txt = txt_file.read()
-            txt_file.close()
-            return txt
+            return output
```

Comment:
What will happen when an error is raised? The returned value is `None` and it doesn't seem like handled.

Reply:
I agree. It appears as if we'll return an empty prediction, not a server error. The caller wouldn't know that a fatal error condition was triggered that cannot be resolved without changing the configuration.